src/sentry/integrations/source_code_management/repository.py

```diff
@@ -170,11 +171,13 @@ def get_stacktrace_link(
                 source_url = self.check_file(repo, filepath, version)
                 if source_url:
                     scope.set_tag("stacktrace_link.used_version", True)
-                    return source_url
+                    return urlquote(source_url, safe="/:?=&")
             scope.set_tag("stacktrace_link.used_version", False)
             source_url = self.check_file(repo, filepath, default)
 
-            return source_url
+            # Encode elements of the filepath like square brackets
+            # Preserve path separators and query params etc.
+            return urlquote(source_url, safe="/:?=&")
```

Comment:
python doesn't seem to have anything comparable to `encodeURI` but this seems close?